### PR TITLE
adopt default style when setting format for date / time

### LINKED ISSUE #1020: Default font with lo_excel->set_default_style doesn't work with dates
If you define a default font using lo_excel->set_default_style and try to write a date with  lo_worksheet->set_cell passing a variable type date in the parameter ip_value, that cell doesn´t use the default font

example report to reproduce the issue.

```
REPORT ztest_excel.
*Based on zdemo_excel36

DATA:
  lo_excel                TYPE REF TO zcl_excel,
  lo_worksheet            TYPE REF TO zcl_excel_worksheet,
  lo_style_arial20      TYPE REF TO zcl_excel_style,
  lv_style_arial20_guid     TYPE zexcel_cell_style,
  lv_date TYPE sy-datum.

CONSTANTS: gc_save_file_name TYPE string VALUE '36_DefaultStyles.xlsx'.
INCLUDE zdemo_excel_outputopt_incl.

START-OF-SELECTION.

  CREATE OBJECT lo_excel.

  lo_style_arial20                  = lo_excel->add_new_style( ).
  lo_style_arial20->font->name      = zcl_excel_style_font=>c_name_arial.
  lo_style_arial20->font->scheme    = zcl_excel_style_font=>c_scheme_none.
  lo_style_arial20->font->size      = 20.
  lv_style_arial20_guid             = lo_style_arial20->get_guid( ).

  lo_excel->set_default_style( lv_style_arial20_guid ).  " Default for all new worksheets

*1st sheet - do not change anything --> defaultstyle from lo_excel should apply
  lo_worksheet = lo_excel->get_active_worksheet( ).
  lo_worksheet->set_title( 'Style for complete document' ).

  lo_worksheet->set_cell( ip_column = 2 ip_row = 4 ip_value = 'All cells in this sheet are set to font Arial, fontsize 20' ).
  lo_worksheet->set_cell( ip_column = 2 ip_row = 5 ip_value = 'because no separate style was passed for this sheet' ).
  lo_worksheet->set_cell( ip_column = 2 ip_row = 6 ip_value = 'but a default style was set for the complete instance of zcl_excel' ).

  lo_worksheet->set_cell( ip_column = 1 ip_row = 10 ip_value = 'test' ).

  lv_date = sy-datum.
  lo_worksheet->set_cell( ip_column = 1 ip_row = 11 ip_value = lv_date ).

  lcl_output=>output( lo_excel ).
```

image

### PR DESCRIPTION
**Pull request by AndreaBorgia-Abo**
first stab at #1020 
must check cloning vs uuid assignment, it works but I'm not convinced this is correct (therefore, draft PR)

### CODE REVIEW

File: src/zcl_excel_worksheet.clas.abap [hunk 1]
```diff
@@ -3685,6 +3685,11 @@
 *                      If not, use default
     DATA: lo_format_code_datetime TYPE zexcel_number_format.
     DATA: stylemapping    TYPE zexcel_s_stylemapping.
+    IF <fs_sheet_content>-cell_style IS INITIAL.
+      DATA(default_style_uuid) = me->excel->get_default_style( ).
```
[the remark this answers is not in the document]
Comment (AndreaBorgia-Abo) [reply src/zcl_excel_worksheet.clas.abap:3689]: @larshp hmm, I'm not sure the suggestion is correct. Should I simply pre-declare the identifier?

**Review by AndreaBorgia-Abo — COMMENTED** [review]
immagine

changes are as expected and the test program works, just a couple of things for the review:
- ~~I do not understand the [linter complaint](https://github.com/abap2xlsx/abap2xlsx/pull/1025/commits/02bb16c71bec7f46d77c6e57530ecf22985c8f0e) about a file I haven't touched @larshp~~ (edit: maybe it is related to the removed class member... never mind: will remove both table and row type)
- I have fixed a silly typo and removed an ununsed class attribute, these should be uncontroversial
- even though it works, I am not convinced this is the right approach: from get_default_style I wanted to derive a zcl_excel_style object, clone it and then assign it (with a new uuid) to the current cell but I couldn't find a way and this is precisely the part on which I would like a second opinion.

**Review by AndreaBorgia-Abo — COMMENTED** [review]
removed unused DDIC objects, should make linter happy

**Comment by sandraros** [comment]
Your fix is good. I'm wondering how to handle this other case, which behaves better with your fix, but still present if we use `SET_DEFAULT_STYLE` late (code to reproduce below):
image

In existing client implementations, maybe they set the default style late. But anyway, your fix is valid, so client who want to use the new feature just need to set the default style early and that's it.

If it's okay for you, I will approve the commit.

PS: code to get screenshot above, after your fix:
```abap
CONSTANTS: gc_save_file_name TYPE string VALUE '36_DefaultStyles.xlsx'.
INCLUDE zdemo_excel_outputopt_incl.
START-OF-SELECTION.
  DATA(lo_excel) = NEW zcl_excel( ).
  DATA(lo_worksheet) = lo_excel->get_active_worksheet( ).
  lo_worksheet->Get_column( 1 )->set_width( 50 ).
  lo_worksheet->set_cell( ip_column = 1 ip_row = 1 ip_value = 'set_cell before set_default_style' ).
  lo_worksheet->set_cell( ip_column = 2 ip_row = 1 ip_value = sy-datum ).
  lo_excel->set_default_style( zcl_excel_style_changer=>create( lo_excel
                    )->set_font_name( zcl_excel_style_font=>c_name_arial
                    )->set_font_scheme( zcl_excel_style_font=>c_scheme_none
                    )->set_font_size( 20
                    )->get_guid( ) ).
  lo_worksheet->set_cell( ip_column = 1 ip_row = 2 ip_value = 'set_cell AFTER set_default_style' ).
  lo_worksheet->set_cell( ip_column = 2 ip_row = 2 ip_value = sy-datum ).
  lcl_output=>output( lo_excel ).
```

**Comment by AndreaBorgia-Abo** [comment]
@sandraros thanks for the code. I'll split my comment in two parts, design and technical.

From a design perspective, I would consider cell A1 to be wrongly formatted and B1 ok, becase at that point in time the default font was still Calibri 11pt, not yet Arial 20pt. [Changing the default font in Excel](https://spreadsheetplanet.com/set-default-font-excel/) requires a restart and applies only to newly created sheets (tried that right now), so the whole point is debatable: what abap2xlsx does here is not how Excel does it, either way, since the setting is global and is applied only upon creation.

From a technical perspective, this means that the default style is accounted for much later in the process when it comes to strings compared to dates. However, considering the design issues above, I'd be happy to let it slide and commit the fix as is.
Thank you :)

**Review by sandraros — COMMENTED** [review]
LGTM 👍

**Pull request merged**